### ISSUE TITLE
Installation of VPS from a container

### ISSUE THREAD

**Issue by edbek** [Ysurac/openmptcprouter-vps#59]
Can i run the installation of VPS from a container (I'm using a docker image Ubuntu-20.04) ? Will this work, or do I need to patch the host kernel by installing the VPS directly from the host itself ?

**Comment by Adorfer** [comment]
technically it's certainly not impossible. but not since the kernel needs to be patched, it will be problably involve some wizzard skills plus with ever update it will generate "backporting workload"

**Comment by edbek** [comment]
Thank you for your reply. Another question: is the "docker-openmptcprouter-vps" project still active or not? It is not clear to me whether it is necessary for this project to patch the kernel to support MPTCP.

**Comment by Ysurac** [comment]
Currently OpenMPTCProuter need a patched kernel in all case. This may change in future release with MPTCP upstreamed in official kernel.

**Comment by edbek** [comment]
I can't run "openmptcprouter-vps" from every container. Need first patch the kernel or take an OS with MPTCP support, and then install "openmptcprouter-vps", but without a kernel patch.
Can you advise on how to get the "openmptcprouter-vps" version without a kernel patch.

Thank you in advance for your response.

**Comment by Ysurac** [comment]
There is no OS that can support current OpenMPTCProuter version. So with current release you need to patch the host.
There is no openmptcprouter-vps options to disable kernel install as it's not supported for now, you have to modify the script yourself.

**Comment by edbek** [comment]
I have an Ubuntu 20.04 operating system installed on my host. Оn the host, I installed the OpenMPTCProuter VPS server of the current version (to patch the kernel) and Docker.
In Docker, a container is running in privileged mode (based on the Ubuntu 20.04 image) and in it I started another separate, isolated OpenMPTCProuter VPS server (there were no errors during its installation into the container).

While the server is running in a container when running the script /usr/bin/omr-admin.py
errors of the form occur:
  sysctl: cannot stat / proc / sys / net / mptcp / mptcp_enabled: No such file or directory
  sysctl: cannot stat / proc / sys / net / mptcp / mptcp_checksum: No such file or directory
  sysctl: cannot stat / proc / sys / net / mptcp / mptcp_path_manager: No such file or directory
  sysctl: cannot stat / proc / sys / net / mptcp / mptcp_scheduler: No such file or directory
  sysctl: cannot stat / proc / sys / net / mptcp / mptcp_syn_retries: No such file or directory
due to the lack of MPTCP system variables in the container (/ proc / sys / net / mptcp)


The kernel is patched on the host,
~ # uname -a
   Linux ed-zabbix 5.4.100-mptcp # 1 SMP Wed Feb 24 09:39:10 UTC 2021 x86_64 x86_64 x86_64 GNU / Linux
and all these variables are present
~ # sysctl net.mptcp
  net.mptcp.mptcp_checksum = 0
  net.mptcp.mptcp_debug = 0
  net.mptcp.mptcp_enabled = 1
  net.mptcp.mptcp_path_manager = fullmesh
  net.mptcp.mptcp_scheduler = blest
  net.mptcp.mptcp_syn_retries = 2
  net.mptcp.mptcp_version = 0

but the container does not.
Is it possible to somehow solve this problem?

Thank you in advance for your response.

**Comment by Adorfer** [comment]
You need to patch the host kernel! 
so you have to get the patches from the regular installer and apply that system. For that you should understand the workings of the installer-script. 
(That is what i meant with my first reply above "wizzard skills needed").

**Comment by mstrolo** [comment]
anyone was able to install it on docker?

**Comment by edbek** [comment]
yes, it managed to be done, but it's not quite a clean way. Anyway, it can be done and it will work.

**Comment by mstrolo** [comment]
can you help me with it?

**Comment by mstrolo** [comment]
@edbek ?

**Comment by Adorfer** [comment]
you can move PARTS of the VPS-functions into a docker container, 
but since you have to patch the host system kernel, i would not consider the installation as "dockerized".

**Comment by dormancygrace** [comment]
Is there any changes? Afaik mptcp is mainlined. Can one use docker or LXC?

**Comment by Ysurac** [comment]
Should be possible with some limitations on host with kernel >=6.6 and with MPTCP enabled.
It's in my internal roadmap.
